### PR TITLE
Add version information

### LINKED ISSUE #71: Add --version option
Add a `--version` option to the quick-lint-js command-line utility which prints quick-lint-js' version number and exits without linting.

Detect `--version` in options.cpp, add a variable to the `option` struct in options.cpp, test the option parsing in test-option.cpp, and print the version message in main.cpp.

See also the `--help` option introduced in PR #20.

### PR DESCRIPTION
**Pull request by spicyyboi**
This adds two flags, --v and --version, along with implementations of two tests in test-options.cpp and therefore closes #71.

As I could not find any version info, I set the version string to "1.0.0", I don't know if that's okay.
I'd be happy to hear your feedback on this!

### CODE REVIEW

**Comment by spicyyboi** [comment]
As some checks weren't happy about me leaving unused version information macros floating around in main.cpp, I removed them.

File: src/main.cpp [hunk 1]
```diff
@@ -37,6 +37,9 @@
 #include <string>
 #include <variant>
 
+// Version information
```
Comment (strager) [inline src/main.cpp:40]: Nit: This comment adds no value IMO. Delete it.

File: src/main.cpp [hunk 2]
```diff
@@ -37,6 +37,9 @@
...
+#define QUICK_LINT_JS_VERSION_STRING "1.0.0"
```
Comment (strager) [inline src/main.cpp:41]: Minor: Let's use `0.1.0` for now.

File: src/main.cpp [hunk 3]
```diff
@@ -304,9 +312,14 @@ void print_help_message() {
   print_option("--output-format=[FORMAT]",
                "Format to print feedback where FORMAT is one of:");
   print_option("", "gnu-like (default if omitted), vim-qflist-json");
+  print_option("--v, --version", "Print version information");
```
Comment (strager) [inline src/main.cpp:315]: Shouldn't it be `-v`, not `--v`, for the short form? (See `-h` and `--help`.)

Actually, it looks like `-h` is broken. O_o Maybe we should make `-v` and `-h` work in a separate commit.

**Review by strager — APPROVED** [review]
Looks great! Let me know when I should merge it.

**Comment by spicyyboi** [comment]
That's good to hear! I've incorporated the first two of your suggestions and I think the pull request can be merged now.

**Comment by strager** [comment]
Landed as commit 2264fe0231ab12ab8a318245c32f9da1d766f152.

**Pull request closed without merging**